### PR TITLE
Fix the issue #24114

### PR DESCRIPTION
**Pull request by Aryan8912**
close: https://github.com/openvinotoolkit/openvino/issues/24114
### Details:
 - *item1*
 - *...*

### Tickets:
 - *https://github.com/openvinotoolkit/openvino/issues/24114*

### CODE REVIEW

**Comment by eshoguli** [comment]
You implemented JIT emitter: good start, thanks! 

I would like to suggest next implementation steps.
1. Update tests to be sure that your implementation is used. Tests should fail: `acl_f32` primitive is used.
2. Update JIT kernel. Tests should fail: expected and reference tensors comparation fails.
3. Finish JIT emitter. Tests should pass.

Let me know if you still have any questions.

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp [hunk 1]
```diff
@@ -309,6 +309,27 @@ class jit_select_emitter : public jit_emitter {
     void emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const;
 };
 
+class jit_softsign_emitter : public jit_emitter {
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp:312]: please rename: `jit_softsign_emitter` => `jit_soft_sign_emitter`

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 2]
```diff
@@ -759,6 +759,49 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+    using TReg = typename dnnl::impl::cpu::aarch64::cpu_isa_traits<isa>::TReg;
+    TReg src = TReg(in_vec_idxs[0]);
+    TReg dst = TReg(out_vec_idxs[0]);
+    TReg aux = TReg(aux_vec_idxs[0]);
+    h->movi(aux.s, 1);
+    h->fdiv(dst.s, src.s, aux.s);
+    h->fadd(dst.s, dst.s, src.s);
+    h->fdiv(dst.s, dst.s, aux.s);
+    h->fsub(dst.s, aux.s, dst.s);
+    h->fmul(dst.s, dst.s, src.s);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:801]: 1. can you align implementation with OpenVINO SoftSign operation from here: https://docs.openvino.ai/2022.3/openvino_docs_ops_activation_SoftSign_9.html
2. you don't need auxiliary register here

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 3]
```diff
@@ -820,6 +820,86 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+    h->fdiv(dst.s, src.s, 1); 
+    h->fadd(dst.s, dst.s, src.s);
+    h->fdiv(dst.s, dst.s, 1); 
+    h->fsub(dst.s, 1, dst.s);  
+    h->fmul(dst.s, dst.s, src.s);
+}
+
+// SoftPlus
+
+jit_softplus_emitter::jit_softplus_emitter(dnnl::impl::cpu::aarch64::jit_generator *host,
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:867]: Move here: https://github.com/openvinotoolkit/openvino/issues/24109, please

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp [hunk 4]
```diff
@@ -335,6 +335,44 @@ class jit_select_emitter : public jit_emitter {
...
+                                                   const std::shared_ptr<ov::Node>& node);
+                                                   size_t get_inputs_count() const override;
+                                                   size_t get_aux_vecs_count() const override;
+                                                   static std::set<std::vector<element::Type>> get_supported_precisions(const std::shared_ptr<ov::Node>& node = nullptr);
+                                                   private:
+                                                   void emit_impl(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const override;
+                                                   template <dnnl::impl::cpu::aarch64::cpu_isa_t isa>
+                                                   void emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const;
+
+};
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.hpp:375]: fix alignment, please

**Comment by Aryan8912** [comment]
@eshoguli I think this is working fine but the problem is a test case does not respond?

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 5]
```diff
@@ -896,6 +896,50 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+    using TReg = typename dnnl::impl::cpu::aarch64::cpu_isa_traits<isa>::TReg;
+    TReg src = TReg(in_vec_idxs[0]);
+    TReg dst = TReg(out_vec_idxs[0]);
+    TReg aux = TReg(aux_vec_idxs[0]);
+    h->movi(aux.s, 1);
+    h->fdiv(dst.s, src.s, 1); 
+    h->fadd(dst.s, dst.s, src.s);
+    h->fdiv(dst.s, dst.s, 1); 
+    h->fsub(dst.s, 1, dst.s);  
+    h->fmul(dst.s, dst.s, src.s);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:938]: Please fix, `SoftSign` implementation: https://docs.openvino.ai/2022.3/openvino_docs_ops_activation_SoftSign_9.html

```
y = x
x = abs(x)
x = x + 1
y = y / x
```

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 6]
```diff
@@ -896,6 +896,51 @@ void jit_select_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
...
+    TReg src = TReg(in_vec_idxs[0]);
+    TReg dst = TReg(out_vec_idxs[0]);
+    TReg aux = TReg(aux_vec_idxs[0]);
+    h->fabs(src.s, src.s);      
+    h->fcvtns(dst.d, src.s);   
+    h->fld1(dst.d, dst.d);    
+    h->fadd(dst.d, dst.d, src.d); 
+    h->flog(dst.d, dst.d);     
+    h->fdiv(dst.d, dst.d, v2.d); 
+    h->fmuls(src.s, dst.s, src.s);
```
Comment (eshoguli) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:939]: don't need it here

Comment (Aryan8912) [reply src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:939]: @eshoguli please provide more detail

**Comment by mlukasze** [comment]
hey @Aryan8912 will you have a time to finish this PR?

**Pull request closed without merging**